src/mongocxx/collection.cpp

```diff
@@ -403,9 +403,21 @@ std::int64_t collection::count(bsoncxx::document::view filter, const options::co
         rp_ptr = options.read_preference()->_impl->read_preference_t;
     }
 
-    auto result = libmongoc::collection_count(
+    // Some options must be added via the options struct
+    bsoncxx::builder::stream::document cmd_opts_builder{};
+
+    if (options.hint()) {
+        auto hint_optional = options.hint();
+        cmd_opts_builder << bsoncxx::builder::stream::concatenate{hint_optional->to_document()};
```

Comment:
would this be better as
`cmd_opts_builder << bsoncxx::builder::stream::concatenate{options.hint()->to_document()};`
then you wouldn't need the `auto` variable